daemon/oci_linux.go

```diff
@@ -665,7 +665,7 @@ func (daemon *Daemon) createSpec(c *container.Container) (*libcontainerd.Spec, e
 	ms = append(ms, c.IpcMounts()...)
 	ms = append(ms, c.TmpfsMounts()...)
 	rootUID, rootGID := daemon.GetRemappedUIDGID()
-	m, err := c.SecretMount(rootUID, rootGID)
+	m, err := c.SecretMount(rootUID, rootGID, daemon.configStore.EnableSecrets)
```

Comment:
can't we check `EnableSecrets` here and just skip `c.SecretMount` if it's `false`?